### ISSUE TITLE
[next/image] Support blob URL's

### ISSUE THREAD

**Issue by sdn90** [vercel/next.js#19291]
# Feature request

## Is your feature request related to a problem? Please describe.

Using a blob currently throws an error.

```
Unhandled Runtime Error
Error: Invalid src prop (blob:http://localhost:3000/ac4ed6bc-8dea-40f9-a6a0-de5b9523a596) on `next/image`, hostname "" is not configured under images in your `next.config.js`
See more info: https://err.sh/next.js/next-image-unconfigured-host
```

## Describe the solution you'd like

It should mimic the behavior of Base 64 encoded images:

https://github.com/vercel/next.js/blob/d47d83d88dc8db43e4b974f8a124794a8e5c9f72/packages/next/client/image.tsx#L239-L243

Not familiar with the codebase but I'm guessing this should work?

```diff
- if (src && src.startsWith('data:')) {
+ if (src && (src.startsWith('data:') || src.startsWith('blob:')) { 
    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs
    unoptimized = true
    isLazy = false
  }
```

## Describe alternatives you've considered
- Convert blob URL to base64
- Reimplementing the code for base64 handling in my component

## Additional context

I'm using Netlify CMS to render a component using `next/image`. It uses blobs to preview an image before persisting it to my storage backend.

**Comment by styfle** [comment]
This has been fixed in [v11.1.1-canary.7](https://github.com/vercel/next.js/releases/tag/v11.1.1-canary.7).

You can try it out today with `yarn add next@canary`, thanks!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
